nibabel/freesurfer/io.py

```diff
@@ -400,6 +525,13 @@ def write_string(s):
             write(len(s))
             write(s, dtype='|S%d' % len(s))
```

Comment:
I think this needs to be:

```Python
s = s.encode() + b'\x00'
write(len(s))
write(s, dtype='|S%d' % len(s))
```

This will correctly handle `str` inputs with `NUL`-termination.

Reply:
Yep, you're right. The `read_annot` code works in both cases though - it seems that `np.fromfile(fobj, '|S%d' % length)` will work regardless of whether the last byte is `\00` or not, as long as the total length is correct.